### PR TITLE
Move SecretGenerator out of Config

### PR DESCRIPTION
**Pull request by bolekk**
SecretGenerator is used to create web session secrets and doesn't really belong to config in any way.
Moving inside Application so it can still be mocked in tests.

### CODE REVIEW

File: core/services/chainlink/application.go [hunk 1]
```diff
@@ -103,6 +103,8 @@ type Application interface {
 
 	// ID is unique to this particular application instance
 	ID() uuid.UUID
+
+	GetSecretGenerator() SecretGenerator
```
Comment (jmank88) [inline core/services/chainlink/application.go:107]: Get/Set prefixes can generally be omitted in Go, although the local style here is aleady a mix so :shrug: 
```suggestion
	SecretGenerator() SecretGenerator
```

**Review by jmank88 — APPROVED** [review]

File: core/services/chainlink/secret_generator_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSecretGenerator_sessionSecret(t *testing.T) {
+	t.Parallel()
+	rootDir := path.Join("/tmp/chainlink_test", "TestSecretGenerator_sessionSecret")
```
Comment (pinebit) [inline core/services/chainlink/secret_generator_test.go:13]: Is this Okay to hardcode the path here? Why not use `ioutil.TempDir`?

File: core/services/chainlink/secret_generator_test.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSecretGenerator_sessionSecret(t *testing.T) {
```
Comment (pinebit) [inline core/services/chainlink/secret_generator_test.go:11]: I think your test `FilePersistedSecretGenerator` specifically, the test name should be `TestFilePersistedSecretGenerator` probably.

File: core/services/chainlink/secret_generator_test.go (hunk 2 above)
Comment (jmank88) [reply core/services/chainlink/secret_generator_test.go:13]: Or https://pkg.go.dev/testing#T.TempDir

**Review by samsondav — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**